Dockerfile.release

```diff
@@ -15,6 +15,15 @@ RUN groupadd -g 1025 pocket && useradd -u 1025 -g pocket -m -s /sbin/nologin poc
 COPY --chown=pocket:pocket release_binaries/pocket_linux_$TARGETARCH /bin/pocketd
 COPY --chown=pocket:pocket tmp/cosmovisor-linux-$TARGETARCH /bin/cosmovisor
 
+# Use MADV_DONTNEED to aggressively return freed memory to the OS
+# This helps reduce RSS and is more transparent in container environments
+ENV GODEBUG="madvdontneed=1"
+
+# Lower GC target percentage (default is 100)
+# Triggers GC when heap grows 50% since last collection
+# Helps control memory growth in production environments
+ENV GOGC="50"
```

Comment:
@Olshansk  IMO we should not build these into the container.
If an operator needs these variables, they can set it in the runtime themselves.

Any objections to remove? @jorgecuesta

Reply:
No, BUT maybe could be adviced, because those enhance the memory management and not so many providers has that knowledge.

Reply:
https://github.com/pokt-network/poktroll/commit/e9bd9547730de72f298108eec7b284171bf8f75d